enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,270 @@
+
+### Open Questions
+
+Currently, the no-config backup sidecar container exits upon error. Since the container is being deployed within the etcd Pod, the whole Pod will never be `Running` as long as the container is failing.
+Once the container fails, it will be automatically restarted according to exponential backoff strategy. This scenario leads to `CrashLoopBackOff` and the etcd Pod will never be `Running`. 
+
+As the no-config backup sidecar is being deployed alongside each etcd member, consequently an error within this container leads to unavailable etcd cluster. Hence, failing the container upon error is not a viable option.
+Also disabling the container from the etcd pod upon error is not recommended, unless requested by the user. As discussed above, enabling and disabling the sidecar container leads to deploying a new version of the static pods.
+This scenario leads to a duration of cluster unavailability. 
+
+As shown above, it is necessary to report the no-config backup sidecar failure without failing the container. Below are possible resolution scenarios. 
+
+- Log the error upon failure
+  - A log parser could be used in order to detect the error, without failing the container.
+  
+- Update Cluster-etcd-Operator StatusConditions
+  - Reporting the condition and degrading the operator could be a solution. 
+
+- Use a Prometheus gauge in order to report the container situation. 
+
+### Implementation Details/Notes/Constraints
+
+The _no-config_ backups builds atop of the existing [automated backup of etcd](https://docs.openshift.com/container-platform/4.15/backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.html#creating-automated-etcd-backups_backup-etcd).
+
+The `PeriodicBackupController` within **cluster-etcd-operator** checks for `Backup` CR with `name=default` within every sync. Upon applying a `Backup` CR with `name=default`, the _no-config_ backup sidecar container is being enabled within each etcd pod in Openshift cluster.
+The `Backup` configurations such as `scheudle`, `timezone` and `retention policy`, are being supplied to the _no-config_ container as flags. 
+According to user's preference, the default configuration could be modified by updating the CR.
+
+The backups are being stored on the local disk of each master node independently, using `hostPath` volume. This replication is required for disaster recovery scenarios where more than one of the master nodes has been lost.
+
+
+A `Backup` CR with default configuration below.
+
+```yaml
+apiVersion: config.openshift.io/v1alpha1
+kind: Backup
+metadata:
+  name: default
+spec:
+  etcd:
+    schedule: "*/5 * * * *"
+    timeZone: "UTC"
+    retentionPolicy:
+      retentionType: RetentionNumber
+      retentionNumber:
+        maxNumberOfBackups: 3
+```
+
+
+
+
+### Risks and Mitigations
+
+## Test Plan
+
+An e2e test will be added to practice the scenario as follows 
+
+- Enable AutomatedBackupNoConfig FeatureGate.
```

Comment:
This is the first mention of this feature gate I'm seeing in this proposal. I don't see any details on this new feature gate and why we are or not using the existing one for automated backups.

Did I miss how you resolved these discussions on using the existing vs a new feature gate?
https://github.com/openshift/enhancements/pull/1646#discussion_r1654846128
https://github.com/openshift/enhancements/pull/1646#discussion_r1711358625
https://github.com/openshift/enhancements/pull/1646#discussion_r1654864573

Reply:
> This is the first mention of this feature gate I'm seeing in this proposal. I don't see any details on this new feature gate and why we are or not using the existing one for automated backups.

Sorry, I have fixed the type, it is the same FeatureGate introduced in https://github.com/openshift/enhancements/blob/master/enhancements/etcd/automated-backups.md

Reply:
> Did I miss how you resolved these discussions on using the existing vs a new feature gate?

I have replied now to https://github.com/openshift/enhancements/pull/1646#discussion_r1711358625

The other two threads have been discussed on slack before, the outcome was to use the same feature gate, to avoid issues when going GA and since it is the same work, using a separate feature gate does not make sense